### PR TITLE
Add guide section for filtering REST collection.

### PR DESCRIPTION
**Pull request by bizley**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #19173

@machour please take a look, feel free to suggest/add anything.

### CODE REVIEW

**Review by samdark — APPROVED** [review]

**Review by Arhell — APPROVED** [review]

File: docs/guide/rest-filtering-collections.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+## Configuring Data Provider For Filtering <span id="configuring-data-provider-for-filtering"></span>
+
+As mentioned in the [Collections](rest-resources.md#collections) section, we can use 
+[Data Provider](output-data-providers#data-providers) to output sorted and paginated list of resources. We can also use 
+it to filter that list.
+
+```php
+$filter = new ActiveDataFilter([
+    'searchModel' => 'app\models\PostSearch'
```
Comment (arogachev) [inline docs/guide/rest-filtering-collections.md:17]:
```suggestion
    'searchModel' => 'app\models\PostSearch',
```

File: docs/guide/rest-filtering-collections.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+[[yii\data\DataFilter::$attributeMap]] to set the map of aliases:
+
+```php
+[
+    'carPart' => 'car_part', // carPart will be used to filter car_part property
+    'authorName' => '{{author}}.[[name]]', // authorName will be used to filter name property of joined author table
+]
+```
+
+## Configuring Filters For ActiveController <span id="configuring-filters-for-activecontroller"></span>
```
Comment (arogachev) [inline docs/guide/rest-filtering-collections.md:168]:
```suggestion
## Configuring Filters For `ActiveController` <span id="configuring-filters-for-activecontroller"></span>
```

Comment (arogachev) [reply docs/guide/rest-filtering-collections.md:168]: Or maybe use a link. :thinking:

**Review by arogachev — CHANGES_REQUESTED** [review]

File: docs/guide/rest-filtering-collections.md (hunk 2 above)
Comment (bizley) [reply docs/guide/rest-filtering-collections.md:168]: I'm not sure how it looks rendered in the guide, also the link is provided in the very first sentence of that paragraph.

Comment (arogachev) [reply docs/guide/rest-filtering-collections.md:168]: Highlighting example can be found here. https://www.yiiframework.com/doc/guide/2.0/en/caching-http#last-modified Not sure about link too. It's not critical. :ok_hand:

**Review by arogachev — APPROVED** [review]

**Pull request merged**